src/figuresClasses.ts

```diff
@@ -1,19 +1,88 @@
-export interface Figure {
+enum Shape {
+  triangle = 'triangle',
+  circle = 'circle',
+  rectangle = 'rectangle',
+}
 
+enum Color {
+  green = 'green',
+  red = 'red',
+  blue = 'blue',
 }
 
-export class Triangle {
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.triangle;
+
+  constructor(
+    public color = Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Invalid value! Please, write another positive number');
+    }
+
+    if (a + b <= c || c + b <= a || c + a <= b) {
+      throw new Error('Incorrect value! Please, write another values');
+    }
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
+    const squareTr: number
+    = Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c));
```

Comment:
do not break line before or after `=`
```suggestion
    const squareTr: number = Math.sqrt(
      p * (p - this.a) * (p - this.b) * (p - this.c)
    );
```